protocol/contracts/libraries/Oracle/LibCurveOracle.sol

```diff
@@ -10,97 +8,105 @@ import "../LibAppStorage.sol";
     function get_balances() external view returns (uint256[2] memory);
 }
 
+/**
+ * @title Oracle
+ * @author Publius, Chaikitty
+ * @notice Tracks the TWAP of the BEAN:3CRV Curve Metapool.
+ */
 library LibCurveOracle {
-    int256 private constant mintPrecision = 100;
+    using SafeMath for uint256;
+    using LibSafeMath32 for uint32;
+    
+    /* Constrains the deltaB to be +/- 1/X of the current Bean supply */
     uint256 private constant MAX_DELTA_B_DENOMINATOR = 100;
 
-    event MetapoolOracle(uint32 indexed season, int256 deltaB, uint256[2] balances);
+    /**
+     * @param season The Season in which the oracle was updated.
+     * @param deltaB The deltaB
+     * @param balances The TWA 
+     */
+    event MetapoolOracle(
+        uint32 indexed season,
+        int256 deltaB,
+        uint256[2] balances
+    );
 
-    using SafeMath for uint256;
-    using LibSafeMath32 for uint32;
+    //////////////////// CHECK ////////////////////
 
     function check() internal view returns (int256 deltaB) {
         deltaB = _check();
         deltaB = checkForMaxDeltaB(deltaB);
     }
 
-    function _check() internal view returns (int256) {
+    function _check() internal view returns (int256 db) {
         AppStorage storage s = LibAppStorage.diamondStorage();
         if (s.co.initialized) {
-            (int256 db, ) = twaDeltaB();
-            int256 mintedSeasons = int256(
-                s.season.current.sub(s.co.startSeason)
-            );
-            mintedSeasons = mintedSeasons > mintPrecision
-                ? mintPrecision
-                : mintedSeasons;
-            return (db * mintedSeasons) / mintPrecision;
+            (db, , ) = twaDeltaB();
         } else {
-            return 0;
+            db = 0;
         }
     }
 
-    function capture() internal returns (int256 deltaB) {
-        deltaB = _capture();
+    //////////////////// CAPTURE ////////////////////
+
+    function capture() internal returns (int256 deltaB, uint256[2] memory balances) {
+        (deltaB, balances) = _capture();
         deltaB = checkForMaxDeltaB(deltaB);
     }
 
-    function _capture() internal returns (int256) {
+    /** 
+     * @dev `balances` stores the TWA balances throughout the Season.
+     * In the case of {initializeOracle}, it will be the current balances.
+     */
+    function _capture() internal returns (int256 deltaB, uint256[2] memory balances) {
         AppStorage storage s = LibAppStorage.diamondStorage();
         if (s.co.initialized) {
-            int256 mintedSeasons = int256(
-                s.season.current.sub(s.co.startSeason)
-            );
-            mintedSeasons = mintedSeasons > mintPrecision
-                ? mintPrecision
-                : mintedSeasons;
-            return (updateOracle() * mintedSeasons) / mintPrecision;
+            (deltaB, balances) = updateOracle();
         } else {
-            initializeOracle();
-            return 0;
+            balances = initializeOracle();
```

Comment:
Use previous balances instead of current balances here for increased manipulation resistance.